### ISSUE TITLE
[Feature request] Support for `eslint` `v9.1.1`

### ISSUE THREAD

**Issue by cewald** [francoismassart/eslint-plugin-tailwindcss#335]
**Is your feature request related to a problem? Please describe.**
I'm using `nuxt` and the latest `@nuxt/eslint` module which utilizes `eslint` version 9 and I couldn't make it work.
If I import the `eslint-plugin-tailwindcss` index file in the new flat-config syntax e.g. an `eslint.config.mjs`, I'll get an error:
```
[Error - 13:47:05] Request textDocument/formatting failed.
  Message: Request textDocument/formatting failed with message: Config (unnamed): Key "rules": Key "classnames-order": Expected severity of "off", 0, "warn", 1, "error", or 2.
  Code: -32603 
```

And here the config file (`eslint.config.mjs`):
```js
// @ts-check
import withNuxt from './.nuxt/eslint.config.mjs'
import pluginVue from 'eslint-plugin-vue'
import pluginTailwindCSS from 'eslint-plugin-tailwindcss'

export default withNuxt([
  ...pluginVue.configs['flat/recommended'],
  pluginTailwindCSS,
  {
    rules: {
      // ...
    }
  }
])
```

**Describe the solution you'd like**
My guess is to [update to the new flat-config](https://eslint.org/docs/latest/extend/plugin-migration-flat-config) syntax or at least describe in the documentation how to properly implement it for the new [configuration syntax](https://eslint.org/docs/latest/use/configure/configuration-files).

**Comment by codeflorist** [comment]
using eslint v8 (which also uses the flat-config), you can do this:

```js
// @ts-check
import { FlatCompat } from '@eslint/eslintrc'
import withNuxt from './.nuxt/eslint.config.mjs'

const compat = new FlatCompat()

export default withNuxt(
	{
		rules: {
			// ...
		}
	}, ...compat.config({
		extends: ['plugin:tailwindcss/recommended'],
		rules: {
			'tailwindcss/no-custom-classname': 'off',
			'tailwindcss/migration-from-tailwind-2': 'off',
		},
	})
)
```

this doesn't seem to work with v9 anymore however. i'd recomment sticking to v8 for now.

**Comment by codeflorist** [comment]
Regarding eslint v9 support:
This comment describes the current source of incompatibility:
https://github.com/eslint/eslint/issues/18405#issuecomment-2084982903
It's the usage of the deprecated `context.parserServices` property.

**Comment by virtuallyunknown** [comment]
Greetings @francoismassart. I installed the `3.15.3-beta.9` and it appears that [every single rule is detecting errors where it should](https://github.com/virtuallyunknown/eslint-flat-config-migration/blob/master/src/tailwind.tsx), the plugin is working as expected.

That said, I am not sure if this is intended or not, but [the recommended config](https://github.com/francoismassart/eslint-plugin-tailwindcss/blob/master/lib/index.js#L24-L42) is still using the eslintrc format, rather than the flat config format. So for example, a config like this won't work out of the box:

```js
import eslintPluginTailwindCss from 'eslint-plugin-tailwindcss';

export const eslintPluginTailwindCssConfig = [
    {
        ...eslintPluginTailwindCss.configs.recommended,
    }
];
```

For me this is not an issue because of [how I configure the plugin](https://github.com/virtuallyunknown/eslint-flat-config-migration/blob/master/configs/tailwind.js), but I thought it might be worth mentioning anyway.

Cheers!

**Comment by cewald** [comment]
I've tried it like that:
```js
import js from "@eslint/js";
import ts from "typescript-eslint";
import tailwind from "eslint-plugin-tailwindcss";

export default [
  js.configs.recommended,
  ...ts.configs.recommended,
  ...tailwind.configs["flat/recommended"],
];
```

It seems, that the `3.15.3-beta.9` version in NPM isn't updated, because the downloaded code is still missing the `flat/recommended` node in the config, [as you can see here](https://www.npmjs.com/package/eslint-plugin-tailwindcss/v/3.15.3-beta.9?activeTab=code) in the `lib/index.js` opposing to the `lib/index.js` [in the GH repos](https://github.com/francoismassart/eslint-plugin-tailwindcss/blob/feat/flat-config-beta/lib/index.js).

It is working if I set it up like this:
```js
//...
  {
    plugins: {
      tailwindcss: pluginTailwindCSS,
    },
    rules: {
      'tailwindcss/classnames-order': 'warn',
      'tailwindcss/enforces-negative-arbitrary-values': 'warn',
      'tailwindcss/enforces-shorthand': 'warn',
      'tailwindcss/migration-from-tailwind-2': 'warn',
      'tailwindcss/no-arbitrary-value': 'warn', // this is disabled in the recommended config
      'tailwindcss/no-custom-classname': 'warn',
      'tailwindcss/no-contradicting-classname': 'error',
      'tailwindcss/no-unnecessary-arbitrary-value': 'warn',
    },
  },
//...
```

**Comment by francoismassart** [comment]
Here is a new version to test,
I need feedback from user using:

- ESLint < v9 + legacy config
- ESLint < v9 + flat config
- ESLint v9 + legacy config
- ESLint v9 + flat config

The beta version to be tested is `eslint-plugin-tailwindcss@3.16.0-beta.1` which can be installed via

`npm i eslint-plugin-tailwindcss@3.16.0-beta.1 -D`

**Comment by cewald** [comment]
Thanks for the update.
I've tested it with ESLint v9 + flat config and it seems to work just fine.

Here is my `eslint.config.mjs`:
```js
//...
import eslintPluginTailwindCSS from 'eslint-plugin-tailwindcss'
import withNuxt from './.nuxt/eslint.config.mjs'

export default withNuxt([
  //...
  ...eslintPluginTailwindCSS.configs['flat/recommended'],
  //...
])

```

**Comment by vladimirsiljkovic** [comment]
Working fine with ESLint 8.57.0 + flat config + Nuxt + AntFu.

`eslint.config.js`:

```js
import antfu from '@antfu/eslint-config'
import eslintPluginTailwindCSS from 'eslint-plugin-tailwindcss'
import withNuxt from './.nuxt/eslint.config.mjs'
//...

export default withNuxt(
  antfu(
    //...
    ...eslintPluginTailwindCSS.configs['flat/recommended'],
    {
      rules: {},
    },
  ),
)
```

**Comment by HugoRCD** [comment]
Working fine with ESLint 9.3.0 + Flat config [repo link](https://github.com/HugoRCD/eslint-config)
```typescript
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-expect-error
import pluginTailwindcss from 'eslint-plugin-tailwindcss'
import type { Linter } from 'eslint'

export default function tailwindcss(): Linter.FlatConfig[] {
  return [
    {
      name: 'tailwindcss/rules',
      plugins: {
        tailwindcss: pluginTailwindcss
      },
      rules: {
        'tailwindcss/classnames-order': 'error',
        'tailwindcss/no-contradicting-classname': 'error',
        'tailwindcss/enforces-shorthand': 'error',
        'tailwindcss/no-custom-classname': 'off',
        'tailwindcss/no-unnecessary-arbitrary-value': 'off',
      }
    }
  ]
}
```

**Comment by virtuallyunknown** [comment]
ESLint v9 + flat config + eslint-plugin-tailwindcss@3.16.0-beta.1 all working good. [Repo](https://github.com/virtuallyunknown/eslint-flat-config-migration)

**Comment by francoismassart** [comment]
Latest version (`3.16.0`) support both flat config and ESLint 9


`npm i eslint-plugin-tailwindcss@3.16.0 -D`

**Comment by cvhau** [comment]
> @francoismassart Thanks for your work!
> 
> Okay, I tested it and had a little hiccup with the different versions `3.16.0-beta.0` & `3.15.3-beta.9` but `3.15.3-beta.9` now seems to work if I use it like this:
> 
> ```js
> import pluginTailwindCSS from 'eslint-plugin-tailwindcss'
> import withNuxt from './.nuxt/eslint.config.mjs'
> 
> export default withNuxt([
>   {
>     plugins: { tailwindcss: pluginTailwindCSS },
>     rules: pluginTailwindCSS.configs.recommended.rules,
>   },
>   // ...
> ])
> ```
> 
> This is the way it is intended to be, right?

Thank you! This config also work for me.
